src/tools/immutable_object_cache/CacheController.cc

```diff
@@ -82,6 +84,9 @@ int CacheController::run() {
       std::bind(&CacheController::handle_request, this,
                 std::placeholders::_1, std::placeholders::_2));
 
+    m_promote_thread = new std::thread([this](){
```

Comment:
I'd prefer to keep this using strictly boost::asio instead of introducing a new thread.